nms/packages/magmalte/app/views/subscriber/SubscriberTable.js

```diff
@@ -161,8 +169,174 @@ async function exportSubscribers(props: ExportProps) {
     });
   }
 }
+const StyledMenu = withStyles({
+  paper: {
+    border: '1px solid #d3d4d5',
+  },
+})(props => (
+  <Menu
+    elevation={0}
+    getContentAnchorEl={null}
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'center',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'center',
+    }}
+    {...props}
+  />
+));
+function SubscriberActionsMenu(props: {onClose: () => void}) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const [open, setOpen] = React.useState(false);
+  const [error, setError] = React.useState('');
+  const enqueueSnackbar = useEnqueueSnackbar();
+  const ctx = useContext(SubscriberContext);
+  const successCountRef = useRef(0);
+  const handleClick = event => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
 
-function SubscribersTable(props: WithAlert & {refresh: boolean}) {
+  const saveSubscribers = async (
```

Comment:
I would suggest restructuring this function in a few ways.

First, chunk the `subscribers` array, and operate on individual chunks, let's call each one `subscribersChunk`. Here's some options to do so: https://stackoverflow.com/questions/8495687/split-array-into-chunks

Next, rather than use a for loop to process each subscriber individually, use `map`, see [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

Also, make sure that `try..catch` only contains the statements that require it. Currently it's wrapping almost everything in the function.